esp/services/esdl_svc_engine/esdl_binding.cpp

```diff
@@ -896,14 +1010,42 @@ EsdlBindingImpl::EsdlBindingImpl(IPropertyTree* cfg,
     }
 }
 
-bool EsdlBindingImpl::reloadDefinitions(IPropertyTree * esdlBndCng)
+void EsdlBindingImpl::saveDESDLState()
 {
-    if ( m_pESDLService)
+    try
+    {
+        //rodrigo is this the best way to fetch the ESDL def for a given service???
+        Owned<IEsdlDefObjectIterator> it = m_esdl->getDependencies(m_espServiceName.str(), "", 0, NULL, 0);
```

Comment:
Ok, so yeah, if we changed the way loadDefinition worked in the firstplace so we only kept a super-set of the dependencies we could just add a IEsdlDefinition::toXML(StringBuffer &) otherwise we have to do something like what is here.